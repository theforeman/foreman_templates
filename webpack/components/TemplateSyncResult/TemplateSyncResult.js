import React from 'react';
import { isEmpty } from 'lodash';
import PropTypes from 'prop-types';

import EmptySyncResult from './components/EmptySyncResult';
import FinishedSyncResult from './components/FinishedSyncResult';

import './TemplateSyncResult.scss';

const TemplateSyncResult = ({
  syncResult: { templates, resultAction, repo, branch, gitUser, pagination },
  history,
  syncedTemplatesPageChange,
  editPaths,
  fileRepoStartWith,
}) => {

  const redirectBack = () => history.push({ pathname: '/template_syncs' });

  return isEmpty(templates) ? (
    <EmptySyncResult primaryAction={redirectBack} />
  ) : (
    <FinishedSyncResult
      templates={templates}
      type={resultAction}
      repo={repo}
      branch={branch}
      gitUser={gitUser}
      fileRepoStartWith={fileRepoStartWith}
      editPaths={editPaths}
      pagination={pagination}
      pageChange={syncedTemplatesPageChange}
    />
  );
};

TemplateSyncResult.propTypes = {
  syncResult: PropTypes.shape({
    templates: PropTypes.array,
    resultAction: PropTypes.string,
    repo: PropTypes.string,
    branch: PropTypes.string,
    gitUser: PropTypes.string,
    pagination: PropTypes.shape({
      page: PropTypes.number,
      perPage: PropTypes.number,
    }),
    syncedTemplatesPageChange: PropTypes.func,
  }).isRequired,
  history: PropTypes.object.isRequired,
  editPaths: PropTypes.object,
  fileRepoStartWith: PropTypes.array,
  syncedTemplatesPageChange: PropTypes.func,
};

TemplateSyncResult.defaultProps = {
  editPaths: {},
  syncedTemplatesPageChange: () => {},
  fileRepoStartWith: [],
};

export default TemplateSyncResult;
