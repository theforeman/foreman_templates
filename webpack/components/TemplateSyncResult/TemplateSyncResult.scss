#foreman-templates {
  .list-view-pf-body {
    .list-view-pf-description {
      width: 20%;
    }
  }

  .list-view-pf-additional-info-item.additional-info-wide {
    width: 15%;
  }

  .list-group-item-heading {
    font-weight: normal;
    font-size: 12px;
  }

  .list-view-header {
    border-top: 0px;
    background-color: #fff;
    border-left-color: #fff;
    border-right-color: #fff;

    .list-view-pf-main-info {
      padding-bottom: 0px;
    }

    .list-group-item-heading {
      font-weight: 600;
    }

    .list-view-pf-left {
      visibility: hidden;
    }
    .list-view-pf-expand {
      visibility: hidden;
      margin-bottom: 0px;
    }
  }
}
