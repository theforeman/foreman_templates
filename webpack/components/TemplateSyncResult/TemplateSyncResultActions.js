import { SYNC_RESULT_PAGINATION_CHANGE } from '../../consts';

export const syncedTemplatesPageChange = pagination => dispatch => {
  return dispatch({ type: SYNC_RESULT_PAGINATION_CHANGE, payload: { pagination } });
}
