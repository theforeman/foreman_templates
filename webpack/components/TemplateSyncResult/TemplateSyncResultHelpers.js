// move to some sort of pagination helper in core
export const templatesPage = (templates, pagination) => {
  const offset = (pagination.page - 1) * pagination.perPage;

  return templates.slice(offset, offset + pagination.perPage);
};
