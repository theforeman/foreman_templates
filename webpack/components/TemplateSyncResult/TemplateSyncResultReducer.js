import Immutable from 'seamless-immutable';
import { deepPropsToCamelCase } from 'foremanReact/common/helpers';

import {
  TEMPLATESYNC_FORM_SUBMITTED,
  SYNC_RESULT_PAGINATION_CHANGE,
} from '../../consts';

export const initialState = Immutable({
  resultAction: '',
  templates: [],

  pagination: {
    page: 1,
    perPage: 20,
  },
});

const syncResult = (state = initialState, action) => {
  const { payload } = action;
  switch (action.type) {
    case TEMPLATESYNC_FORM_SUBMITTED:
      return state.merge({
        ...deepPropsToCamelCase(payload.data),
      });
    case SYNC_RESULT_PAGINATION_CHANGE:
      return state.set('pagination', payload.pagination);
    default:
      return state;
  }
};

export default syncResult;
