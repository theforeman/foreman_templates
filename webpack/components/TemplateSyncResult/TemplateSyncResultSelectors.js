export const selectSyncResult = state => state.foremanTemplates.syncResult;
