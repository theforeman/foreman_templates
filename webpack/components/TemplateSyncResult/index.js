import { connect } from 'react-redux';

import TemplateSyncResult from './TemplateSyncResult';
import * as TemplateSyncResultActions from './TemplateSyncResultActions';

import { selectSyncResult } from './TemplateSyncResultSelectors';

const mapStateToProps = state => ({ syncResult: selectSyncResult(state) });

export default connect(
  mapStateToProps,
  TemplateSyncResultActions
)(TemplateSyncResult);
