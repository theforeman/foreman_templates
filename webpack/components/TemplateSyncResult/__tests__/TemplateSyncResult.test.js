import { testComponentSnapshotsWithFixtures } from 'react-redux-test-utils';

import {
  importTemplates,
  exportTemplates,
} from '../__fixtures__/templateSyncResult.fixtures';
import TemplateSyncResult from '../TemplateSyncResult';

const fixtures = {
  'should render export result': {
    syncResult: {
      resultAction: 'export',
      templates: exportTemplates,
      repo: 'https://github.com/theforeman/community-templates.git',
      branch: 'master',
      gitUser: 'Admin',
    },
    history: {},
  },
  'should render empty state when no templates synced': {
    syncResult: {
      templates: [],
    },
    history: {},
  },
  'should render import result': {
    syncResult: {
      resultAction: 'import',
      templates: importTemplates,
      repo: '/home/vagrant/templates',
    },
    history: {},
  },
};

describe('TemplateSyncResult', () =>
  testComponentSnapshotsWithFixtures(TemplateSyncResult, fixtures));
