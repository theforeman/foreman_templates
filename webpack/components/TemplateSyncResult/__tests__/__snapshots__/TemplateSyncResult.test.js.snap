// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TemplateSyncResult should render empty state when no templates synced 1`] = `
<EmptySyncResult
  primaryAction={[Function]}
/>
`;

exports[`TemplateSyncResult should render export result 1`] = `
<FinishedSyncResult
  branch="master"
  editPaths={Object {}}
  fileRepoStartWith={Array []}
  gitUser="Admin"
  pageChange={[Function]}
  pagination={Object {}}
  repo="https://github.com/theforeman/community-templates.git"
  templates={
    Array [
      Object {
        "additionalErrors": null,
        "className": "JobTemplate",
        "errors": Object {},
        "humanizedClassName": "Job Template",
        "locked": true,
        "name": "random template",
        "snippet": true,
        "templateFile": "random_template.erb",
      },
      Object {
        "additionalErrors": null,
        "className": "Ptable",
        "errors": Object {},
        "humanizedClassName": "Ptable",
        "locked": false,
        "name": "CoreOS default",
        "snippet": false,
        "templateFile": "coreos_default.erb",
      },
    ]
  }
  type="export"
/>
`;

exports[`TemplateSyncResult should render import result 1`] = `
<FinishedSyncResult
  branch=""
  editPaths={Object {}}
  fileRepoStartWith={Array []}
  gitUser=""
  pageChange={[Function]}
  pagination={Object {}}
  repo="/home/vagrant/templates"
  templates={
    Array [
      Object {
        "additionalErrors": "No \\"name\\" found in metadata",
        "errors": null,
        "name": null,
        "templateFile": "random_template.erb",
      },
      Object {
        "additionalErrors": null,
        "className": "ProvisioningTemplate",
        "errors": Object {
          "base": Array [
            "This template is locked",
          ],
        },
        "humanizedClassName": "Provisioning Template",
        "kind": "Provision",
        "locked": true,
        "name": "EPEL",
        "snippet": true,
        "templateFile": "epel.erb",
      },
      Object {
        "additionalErrors": null,
        "className": "Ptable",
        "errors": Object {},
        "humanizedClassName": "Ptable",
        "locked": false,
        "name": "CoreOS default",
        "snippet": false,
        "templateFile": "coreos_default.erb",
      },
      Object {
        "additionalErrors": "No \\"model\\" found in metadata",
        "errors": null,
        "name": null,
        "templateFile": "no_model.erb",
      },
      Object {
        "additionalErrors": null,
        "className": "ProvisioningTemplate",
        "errors": Object {
          "name": Array [
            "has already been taken",
          ],
        },
        "humanizedClassName": "Provisioning Template",
        "locked": true,
        "name": "Kickstart fake",
        "snippet": true,
        "templateFile": "kickstart_fake.erb",
      },
    ]
  }
  type="import"
/>
`;
